modules/wp-seo/wp-seo-front.php

```diff
@@ -38,9 +50,87 @@ function qtranxf_wpseo_webpage_schema( $piece, $context ) {
     add_filter( 'wpseo_breadcrumb_single_link_info', 'qtranxf_wpseo_breadcrumbs_link', 10, 3 );
     function qtranxf_wpseo_breadcrumbs_link( $link_info, $index, $crumbs ) {
         $link_info['text'] = qtranxf_useCurrentLanguageIfNotFoundUseDefaultLanguage( $link_info['text'] );
+        $link_info['url']  = qtranxf_convertURL( $link_info['url'], $lang );
 
         return $link_info;
     }
+
+    function qtranxf_wpseo_schema_website( $data ) {
+        $data['@id'] = qtranxf_convertURL( $data['@id'], $lang );
+        $data['url'] = qtranxf_convertURL( $data['url'], $lang );
+
+        return $data;
+    }
+
+    add_filter( 'wpseo_schema_website', 'qtranxf_wpseo_schema_website' );
+
+    function qtranxf_wpseo_json_ld_search_url( $search_url ) {
+        $search_url = qtranxf_convertURL( $search_url, $lang );
+
+        return $search_url;
+    }
+
+    add_filter( 'wpseo_json_ld_search_url', 'qtranxf_wpseo_json_ld_search_url', 10, 3 );
+
+    function qtranxf_wpseo_schema_imageobject( $data ) {
+        $data['@id'] = qtranxf_convertURL( $data['@id'], $lang );
+
+        return $data;
+    }
+
+    add_filter( 'wpseo_schema_imageobject', 'qtranxf_wpseo_schema_imageobject' );
+
+    function qtranxf_wpseo_schema_webpage( $data ) {
+        $data['@id'] = qtranxf_convertURL( $data['@id'], $lang );
+        $data['url'] = qtranxf_convertURL( $data['url'], $lang );
+        if ( $data['isPartOf']['@id'] ) {
```

Comment:
Better test with `isset` or `! isempty`. Do you know what is needed? I suppose `isset`, but not sure.

Reply:
Yes, better to test with `isset`.

Reply:
Done.